platform/email/src/main/java/org/motechproject/email/settings/SettingsDto.java

```diff
@@ -46,32 +64,64 @@ public Properties toProperties() {
         Properties properties = new Properties();
         properties.put(MAIL_HOST_PROPERTY, host);
         properties.put(MAIL_PORT_PROPERTY, port);
+        properties.put(MAIL_USERNAME_PROPERTY, username);
+        properties.put(MAIL_PASSWORD_PROPERTY, password);
         properties.put(MAIL_LOG_ADDRESS_PROPERTY, logAddress);
         properties.put(MAIL_LOG_SUBJECT_PROPERTY, logSubject);
         properties.put(MAIL_LOG_BODY_PROPERTY, logBody);
         properties.put(MAIL_LOG_PURGE_ENABLE_PROPERTY, logPurgeEnable);
         properties.put(MAIL_LOG_PURGE_TIME_PROPERY, logPurgeTime);
         properties.put(MAIL_LOG_PURGE_TIME_MULTIPLIER_PROPERTY, logPurgeTimeMultiplier);
-
         return properties;
     }
 
-    public SettingsDto(String host, String port, String logAddress, String logSubject, String logBody, String logPurgeEnable) {
-        this.host = host;
-        this.port = port;
-        this.logAddress = logAddress;
-        this.logSubject = logSubject;
-        this.logBody = logBody;
-        this.logPurgeEnable = logPurgeEnable;
-        this.logPurgeTime = "0";
-        this.logPurgeTimeMultiplier = "0";
+    public SettingsDto(String host, String port, String user, String password, InputStream additionalProperties, // NO CHECKSTYLE More than 7 parameters (found 8).
+                       String logAddress, String logSubject, String logBody, String logPurgeEnable,
+                       String logPurgeTime, String logPurgeTimeMultiplier) {
+        this(host, port, user, password, new HashMap<String, String>(), logAddress, logSubject, logBody, logPurgeEnable, logPurgeTime, logPurgeTimeMultiplier);
+        String data;
+        try {
+            data = IOUtils.toString(additionalProperties);
+            data = data.replaceAll("\\{", "");
+            data = data.replaceAll("}", "");
+            if (!"".equals(data)) {
+                for (String property : data.split(",")) {
+                    String[] p = property.replaceAll(" ", "").split("=");
+                    if (p.length == 2) {
+                        this.additionalProperties.put(p[0], p[1]);
+                    } else {
+                        throw new MotechException("Error loading raw file config to properties");
+                    }
+                }
+            }
```

Comment:
Seems like all of this can (and should) be replaced with load(InputStream is) from Properties class.